### ISSUE TITLE
Quest for type of building

### ISSUE THREAD

**Issue by Monduiz** [westnordost/StreetComplete#25]
I am very impressed with this app. I have a couple of questions for you.

Is it possible to have a quest for the type of buildings? This is one of the most missing information about buildings. Could a quest be added for mobility access too?

**Comment by westnordost** [comment]
Yes and yes. For mobility access, please create a separate ticket.

The problem (and the reason why I haven't implemented it yet) is that there are is a heap of different building types. So I don't how I can best find the balance between keeping it simple and clear and enabling the user to always give a precise and correct anser (= presenting him all the options).

**Comment by Monduiz** [comment]
I certainly see what you mean. It is not an easy design decision.

Would a scrolling menu work well to choose a tag? iOS has the tumbler.

**Comment by westnordost** [comment]
Well I thought about something like the roof selector (pictures or pictures with text), only grouped by category: Residential, Commercial, industrial etc.

I am doing the same now with street surfaces. But perhaps not everything fits into categories like that (is a Church, a Hospital,... residential or commercial?)

**Comment by Monduiz** [comment]
Yes, buildings can have the value of house, apartments, church, temple, hospital, university, and more. All these are valid for "building=*". 

Maybe they could be categorized further for the UI: 

Education -> school, university;
Worship -> church, shrine, temple, synagogue, mosque, cathedral, chapel
Residential -> residential, house, apartments, bungalow, static_caravan;
Commercial -> commercial, hotel, bakehouse, etc.
Industrial -> industrial, farm, 
storage-distribution -> warehouse
Other -> civic, hospital, train_station, transport, etc.

The objective would be to channel users through the complexity of these tags and there intended use. Not an easy one but this could be brainstormed!

**Comment by johnwhelan** [comment]
Currently looking in Ottawa I think the quest looks at building levels.  

Could different quests do different things?

If so then simple things can be given to new mappers but more complex things such as building type to more experienced mappers.  ie you have to have used something for more than 30 minutes?

**Comment by westnordost** [comment]
Yes of course. This ticket is about adding a new quest where the user specifies the building type.

**Comment by NonnEmilia** [comment]
For this specific quest note that the `building=*` key is intended (mainly) for the building typology rather than the building usage.

For example `building=farm` is a typical residential building inside a farm, not an "industrial" one.

See [Key:building](https://wiki.openstreetmap.org/wiki/Key:building) (with its [talk](https://wiki.openstreetmap.org/wiki/Talk:Key:building#Scope)) and [Buildings](https://wiki.openstreetmap.org/wiki/Buildings) on the Wiki.

**Comment by zxmon21** [comment]
I would like to see this quest implemented

**Comment by matkoniecz** [comment]
From #297 

> If you want to push towards that #25 is implemented (it will be sooner or later), you can contribute by searching for good and representative pictures for all the buildings. The ones in the wiki are not always good and generic enough to be representative (worldwide).

Is there a list of required photos (it would not be really useful to search for ones that already have a good photo...)?

**Comment by Binnette** [comment]
Hi, here is my mockup for "[accommodation](https://wiki.openstreetmap.org/wiki/Key:building#Accommodation)" buildings (remarks are welcome) :
![accommodation](https://user-images.githubusercontent.com/6597721/26902691-850ef13e-4bda-11e7-887d-db9e1b2af08e.png)

**Comment by westnordost** [comment]
The icons are well thought through, I would probably understand what is meant without the text. I am not sure though if the "average user" which doesn't know the OSM categories would. (Though I do not understand "house". [Doppelhaushälfte](https://www.google.de/search?q=doppelhaush%C3%A4lfte&safe=off&source=lnms&tbm=isch&sa=X&ved=0ahUKEwjHvLif3KzUAhXDJ8AKHVjyBJMQ_AUIBigB&biw=1132&bih=919)?)
In my mind, I sketched a slightly different view: Instead of quadratic icons in a grid, have a list of items which each have an icon/photo on the left and a sentence of explanation that fills the right side because after reading the wiki page, I had the impression that each category needs some explanation.

Also, why not photos?

**Comment by Binnette** [comment]
I choosed icons to get rid of architectural and cultural aspects. I think icons are understandable by more people. In my concern, pictures are less "readable". :smile:

**Comment by Binnette** [comment]
And yes, I do not really understand the difference between house and detached. The wiki is not very precise.

**Comment by HolgerJeromin** [comment]
I really like the icons. Don't think photos would work in this size. 
=house is not well named, but I also use them for buildings which share a wall with one other residential building.

**Comment by krzyk** [comment]
Nici icons but the picture for house looks like a one for semi-detached house (which is missing and those are pretty common in Poland). During my mapping I use building=house as a more specialized version of building=yes, that is a place where people live.

**Comment by HolgerJeromin** [comment]
@krzyk What is the tag for a semi-detached house in your opinion? 

According to the [building wiki](https://wiki.openstreetmap.org/wiki/Key:building) page =house is semi-detached (aka two family side by side building). But according to the [=house](https://wiki.openstreetmap.org/wiki/Tag:building%3Dhouse) wiki it is a generic one family building but could also be semi-detached. 8-/

The generic "living house" (regardless the amount of families) is not =house but =residential...

**Comment by NonnEmilia** [comment]
I think there are two level of generalization:
1. Residential: including apartments, terrace, houses, ...
2. House: including detached, farm, ...

Also `building=house` is another way to tag `building=terrace` because `building=terrace` is for the entire row of houses, but you could not tag them as `building=terrace` but as many single `building=house`s attached to each other.

As written in the wiki `building=house` "should share at least two nodes with joined neighbours". Not so simple to explain it with an image... 😞

**Comment by westnordost** [comment]
Because of this unclear definition, I have always been mapping any kind of building where one family has its own living unit (so probably house, detached, semi-detached, terrace) simply as house. I think it is quite common, at least in my area, to use house for what would probably actually be detached.

This is probably because "house" is used in actual lamguage and everyone knows what is meant with it, noone says "I am going back to my detached".

Also, the definition "should share at least two nodes with another house" is shit. If this is supposed to be the difference bezween detached/semi-detached and house, then this is duplicate information: That info is already in the geometry!

All in all, I am for only showing house, the other stuff is to confusing. Not even for us OSMlers it is clear.

Am 8. Juni 2017 08:57:57 MESZ schrieb NonnEmilia <notifications@github.com>:
>I think there are two level of generalization:
>1. Residential: including apartments, terrace, houses, ...
>2. House: including detached, farm, ...
>
>Also `building=house` is another way to tag `building=terrace` because
>`building=terrace` is for the entire row of houses, but you could not
>tag them as `building=terrace` but as many single `building=house`s
>attached to each other.
>
>As written in the wiki `building=house` "should share at least two
>nodes with joined neighbours". Not so simple to explain it with an
>image... 😞
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/westnordost/StreetComplete/issues/25#issuecomment-307016542

-- 
Diese Nachricht wurde von meinem Android-Mobiltelefon mit K-9 Mail gesendet.

**Comment by Binnette** [comment]
So, maybe it's better like this. With 2 groups :

1. Residential: Apartments; Dormitory; Hotel
2. House: Detached; Farm; Terrace; Houseboat; Bungalow; Static caravan

Note that, user can also select Residential or House.

![accommodation b](https://user-images.githubusercontent.com/6597721/26927602-e7e3d842-4c52-11e7-89a1-9f300862dae3.png)

Here is an other proposition for group "House" with a question mark :
![house](https://user-images.githubusercontent.com/6597721/26927833-c7ff7102-4c53-11e7-8645-fe1a1edf99ca.png)

**Comment by NonnEmilia** [comment]
Another compromise is to exclude "Detached" from the options.
`building=detached` may create confusion with `building=house`, but according to the current schema `building=house` is a superset  of `building=detached` so there's no data loss and the result is more [KISS](https://en.wikipedia.org/wiki/KISS_principle).

If anyone want to be more specific and tag detached houses as `building=detached` will use an advanced OSM editor.

**Comment by johnwhelan** [comment]
Detached is a single family home in North America.  House is either detached or semi-detached.  I would keep the distinction.  Detached means detached, house is a more general term that includes detached so dropping detached would mean loss of detail.

John

**Comment by NonnEmilia** [comment]
I totally agree with you in fact I said that is a compromise, a compromise
between details and simplicity.

I am proposing to keep a little level of details in this app (and don't
have only an "House" button) and delegate the more complex detail level to
any other more complete OSM editor (JOSM, iD, ...).

I personally like the actual simplicity of StreetComplete.

2017-06-08 18:25 GMT+02:00 johnwhelan <notifications@github.com>:

**Comment by johnwhelan** [comment]
But in Canada Stats Canada is helping feed building information into OSM
and they'd like as much detail as they can get.  If streetcomplete is
unable to provide the level of detail they would like then they will look
for other solutions.  Currently much of Ottawa has had details added to
buildings using streetcomplete and they are looking to expand across Canada.

Some Montreal economists are very excited about the building information
and using it.  Zambia local government is also looking at it.

Stats is using what I call non-mappers to gather its building data but
wants the data to be as rich as possible.

We don't normally think about what the data in OSM will be used for but in
this case it is worth thinking about.  The value is there.

Cheerio John

**Comment by westnordost** [comment]
The problem with the second proposal is that "residential" is clearly a category which can be made more detailed while house is not necessarily a category (right?) but that setup would suggest that it is.
Anyway, house is clearly a subcategory of residential, having it alongside residential seems wrong.

There is always the possibility to create another quest to let the user specify "what kind of house" something is. Offering too many choices at once will only confuse the user.

**Comment by Binnette** [comment]
Here is a possible way to solve the problem. There are 56 values for tag building. Here is my current "categorization". Remarks are welcome.

So i am thinking of a screen that shows 7 buttons : Residential; Services/Civic; Religious; Commercial; For Cars; Farm; Sport; Other

When you tap on one of that 7 buttons, a specific popup is open and you can choose between tags.

![tab](https://user-images.githubusercontent.com/6597721/26953249-8dabcc84-4ca9-11e7-8084-a8e66a920ba1.png)

**Comment by matkoniecz** [comment]
I would expect hotel in services.

**Comment by matkoniecz** [comment]
Also, I am not entirely sure whatever horrible building=ruins tagging scheme deserves to be promoted.

**Comment by westnordost** [comment]
If you think that all those should fit into the quest, I had begun working on a View that might facilitate this (but then had other things to do) originally for the street surface quest. I can push the branch this evening to origin if I remember.
The concept is basically https://github.com/westnordost/StreetComplete/issues/76#issuecomment-290947791 (but perhaps instead of sliding, there is a zoom-into animation)

**Comment by matkoniecz** [comment]
I tried to make mockup for navigating this list of 56 values, but after some thinking it seems that differentiating between various residential values seems to be highly complicated and of little use.

I propose to simply show following in a list

- residential
- garage
- garages
- industrial
- commercial
- retail

With "more" button listing also:

- shed
- roof
- school
- retail
- greenhouse
- church
- farm_auxiliary
- transportation

It would allow correct answer without creating note in more than 97% of quests.

Various values for residential are poorly defined complicated mess with dubious usefulness, so using just building=residential to keep everybody sane is a good idea.

There is potential to make separate quest for making building=residential more details.

Top values not covered by above list: building=construction, =service, =civic, =office, =university, =collapsed (invalid tagging), =public (deprecated), =hospital, =hotel, =entrance (deprecated), =kindergarten, =ruins (invalid tagging), =damaged (invalid tagging), commercial;residential (invalid), -college, =no (???), =bunker =pajaru (unclear meaning)

**Comment by johnwhelan** [comment]
>differentiating between various residential values seems to be highly complicated and of little use.

There are end users who would strongly disagree with you.  Quite a few municipal governments can make use of this information.

John

**Comment by matkoniecz** [comment]
> can make use of this information

Still, it is so complicated that it would be better to split this into a separate quest. BTW, do you have a link to description how this data was used? I am quite curious.

**Comment by johnwhelan** [comment]
Statistics Canada tend to slice and dice it and combine it with other data for statistical studies.  Their project is current and is more in the data collection phase at the moment but there are lots of people showing interest including a group of economists in Montreal.

**Comment by chrisdebian** [comment]
Hi, all.

This looks like a really good idea for a quest.

'Still, it is so complicated that it would be better to split this into a separate quest'.

Any thoughts on which separate quests should be considered?

Cheers,

Chris.

**Comment by openbrian** [comment]
I agree with @matkoniecz .  Instead of jumping from building=yes to building={1 of 56 types}, perhaps StreetComplete can do this in stages.  The first quest picks the type {residential, service, commercial, other}.  Granted there is no building=civic, but StreetComplete can use a tag like streetcomplete:building=civic (or is it building:streetcomplete=civic).  This would trigger a new services quest.  Same for religious, and the other categories.

At the top level, this gives you room for construction, which I bumped into today.

**Comment by krzyk** [comment]
@westnordost Have you thought about @matkoniecz idea to mark it first as general category building and have other quests to make e.g. residential into a house or apartments?

Or you would prefer to make this quest about the top 98% building types like you mentioned:
> house (51%), residential (16%), garage or apartments (each 5%).
This way it would be a two-click to mark the above building types (compare it with surface quest for asphalt, where it is click, scroll, click and click to confirm).


In the later case, if the building is of other type user would need to make a note, right? Or maybe like it is for the roof shape quest, where additional types are accessible by scrolling and scrolling+click?

**Comment by PanderMusubi** [comment]
Here are also some SVG icons you can use in this context http://www.sjjb.co.uk/mapicons/contactsheet (these are also available in Inkscape Open Symbols).

**Comment by westnordost** [comment]
@matkoniecz  @krzyk @openbrian In theory, a good idea. But for many of these categories, a documented tag is missing. I wouldn't want to create a `:streetcomplete` namespace nor invoke the wrath of people watching over building-tag-values.

**Comment by bitboy85** [comment]
By now, there seems to be another issue with buildings. I get no quests about housenumbers if the tagging is only "building=yes". And i've seen lots of them by now so this is an important quest.
To Keep it simple but correct you can add "simple" or "advanced" view of possible answers.
simple just contains the most common ones (if ist wrong, that could also happen if a user directly edit osm and don't know all possible taggings). The advanced view contains all possible values with easy to understand icons like the ones from Binnette.

**Comment by rugk** [comment]
What you mean, might be an "expert mode" as requested in https://github.com/westnordost/StreetComplete/issues/471.

**Comment by westnordost** [comment]
**Currently my opinion about how this quest type should be implemented is this:**

Simply show a plain list with images (like for the roof shapes quest) with the most-used tags for `building`, sorted by likeliness ("house" etc at the top).

For all the less frequent values (at some point we need to make a cut), the user will have to leave a note. There could be a "Other answer" option named "Other kind of building" in which the user has to describe the kind of building and that does actually leave a note.

Though, I am still unsure on how to deal with the odd situation with "categories" that are not really categories because not all values are categorized (see https://github.com/westnordost/StreetComplete/issues/25#issuecomment-307243727) and for others, no generic tag for that category exists.

**Comment by openbrian** [comment]
This is a great point.  In OSS we often declare all options equally important, but realistically there are only a few building types that make up the majority of buildings out there.

**Comment by afmenez** [comment]
My worry is that a two level interface like the one for road surface would be too cumbersome.  It's OK for roads, where you have to select once in a while during your walk.  But for buildings, where you will probably have to select a new one every few steps, the interface should be minimal.  Maybe 3 or 4 most common building types on the first level, with an "other" button for churches, schools and so on.

**Comment by rugk** [comment]
Or you just place the top-3 at the top, so optionally a user can still select the pother types.

**Comment by westnordost** [comment]
The main problem here is what to do if the user cannot specify tge type of building beyond "it is a building" - i.e. it may be a commercial building or also a residential building or even just a retail building etc...

Am 21. Dezember 2017 00:49:48 GMT+07:00 schrieb rugk <notifications@github.com>:
>Or you just place the top-3 at the top, so optionally a user can still
>select the pother types.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/westnordost/StreetComplete/issues/25#issuecomment-353134626

-- 
Diese Nachricht wurde von meinem Android-Mobiltelefon mit K-9 Mail gesendet.

**Comment by padan** [comment]
There should be a quickly and directly accessible icon for the most frequent types like "house" and "garage" (to easily convert building=yes into building=house which require housenumbers or building=garage which does not require housenumbers) and a button "other" which invokes a submenu, with other frequently used building types with icons and names (similar to sports types).

**Comment by ENT8R** [comment]
Yes, I am implementing the quest like this now!

**Comment by RubenKelevra** [comment]
Interesting conversations about building types. I guess this app has the ability to spread much more detailed information throughout the database, so the first implementation should be very well to understand, to avoid mass misstagging.

I see some issues here for example in residential buildings, build in a row inside a city with shops in the first level. How should such a building be tagged? 🤔

From a user perspective this opens at least 5 possibilities: residential, house, commercial, retail, terrace. So each user might select different on their view onto the world.

I think we should approach it differently: we collect pictures of buildings as a crowd and agree on each of them how we should tag them. So we might end up with some more good pictures than values, but completely cover all edge-cases.

**Comment by ENT8R** [comment]
Have you noticed #774? This should answer most of your proposals...

> so the first implementation should be very well to understand, to avoid mass misstagging.

This should be the intention behind **every** quest 👍 

> From a user perspective this opens at least 5 possibilities

which is why we decided to use icons rather than images and a show a small description next to the icon like you can see in https://github.com/westnordost/StreetComplete/pull/774#issuecomment-380835490

**Comment by RubenKelevra** [comment]
Thanks for your work on this topic, I actually overlooked this PR here in the list. Thanks for the heads-up

**Comment by Nielkrokodil** [comment]
Your app shows for building type = semidetached_house the whole building (2 households) as one object. 

This ist not what the wiki and osm.org think it is. They interpret this tag as "half of the semidetached house" or "house attached to another house on one side"  (german "Doppelhaushälfte" statt "Doppelhaus"). 
https://wiki.openstreetmap.org/wiki/Tag:building=semidetached%20house?uselang=de

**Comment by westnordost** [comment]
1. The app can only be as clear as the wiki and the wiki is not very clear about it, see the discussion page. I bet if you initiated a discussion on it on the forum or mailing list, you'll both find people who argue for Doppelhaus and people who argue for Doppelhaushälfte.
2. wiki articles in other languages than English should not deviate from the English wiki page but basically be a translation
3. if the definition has always been that unclear, then this definition becomes the de-facto definition over time. In other words, some semi-detached houses are Doppelhaushälften, some are Doppelhäuser.
4. In any case, this does not concern StreetComplete that much because the building geometry has already been added. Only the type of building is added by StreetComplete. So if two buildings were "wrongfully" mapped as one, the mistake is already there and StreetComplete (users) don't touch that. On the contrary, by tagging it as `semidetached_house`, they give the necessary hint what kind of building in order to "fix" it (one way or another)

**Comment by Nielkrokodil** [comment]
Thank you! I prefer your Doppelhaus-interpretation anyway :)

**Comment by westnordost** [comment]
(me too, because the Doppelhaushälfte interpretation would not be that compatible with simple 3d buildings. According to simple 3d buildings, anything that shares a wall and at least things that share a roof should be one `building`, everything else is `building:part`. Also, Doppelhaus-interpretation is consistent with `building=terrace`)
